Example/PrimerSDK/MerchantCheckoutViewController.swift

```diff
@@ -229,94 +242,159 @@ extension MerchantCheckoutViewController: PrimerDelegate {
-        
+
         callApi(request) { (result) in
             switch result {
             case .success(let data):
-                var paymentResponse: PaymentResponse?
+//                var paymentResponse: PaymentResponse?
+
                 if let dic = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any] {
                     if let amount = dic?["amount"] as? Int,
                        let id = dic?["id"] as? String,
                        let date = dic?["date"] as? String,
-                       let statusStr = dic?["status"] as? String,
-                       let status = PaymentStatus(strValue: statusStr) {
-                        
-                        var requiredAction: RequiredAction?
-                        
+                       let status = dic?["status"] as? String {
+
                         if let requiredActionDic = dic?["requiredAction"] as? [String: Any] {
-                            if let requiredActionNameStr = requiredActionDic["name"] as? String,
-                               let requiredActionName = RequiredActionName(strValue: requiredActionNameStr),
-                               let description = requiredActionDic["description"] as? String {
-                                requiredAction = RequiredAction(name: requiredActionName,
-                                                                description: description,
-                                                                clientToken: requiredActionDic["clientToken"] as? String)
+                            self.transactionResponse = TransactionResponse(id: id, date: date, status: status, requiredAction: requiredActionDic)
+                            
+                            if let requiredActionName = requiredActionDic["name"] as? String,
+                               let clientToken = requiredActionDic["clientToken"] as? String {
+
+                                if requiredActionName == "3DS_AUTHENTICATION", status == "PENDING" {
+                                    resumeHandler?.handle(newClientToken: clientToken)
+                                    return
+                                }
                             }
                         }
-                        
-                        paymentResponse = PaymentResponse(amount: amount, id: id, date: date, status: status, requiredAction: requiredAction)
                     }
                 }
-                
-                if paymentResponse != nil {
-                    Primer.shared.receivedPaymentResponse(paymentResponse!, for: paymentMethodToken)
-                } else {
-                    completion(nil)
-                }
-                
+
+                resumeHandler?.handleSuccess()
+
             case .failure(let err):
-                completion(err)
+                resumeHandler?.handle(error: err)
             }
         }
     }
     
+    func tokenAddedToVault(_ token: PaymentMethodToken) {
```

Comment:
Do we need `resumeHandler` here?

Reply:
We haven't agreed on adding `resumeHandler` here, right? Moreover, the `onTokenizationSuccess` will be called anyway.